redis/commands/graph/query_result.py

```diff
@@ -360,3 +360,166 @@ def cached_execution(self):
     @property
     def run_time_ms(self):
         return self._get_stat(INTERNAL_EXECUTION_TIME)
+
+
+class AsyncQueryResult(QueryResult):
+    def __init__(self):
+        pass
+
+    async def initialize(self, graph, response, profile=False):
+        self.graph = graph
+        self.header = []
+        self.result_set = []
+
+        # in case of an error an exception will be raised
+        self._check_for_errors(response)
+
+        if len(response) == 1:
+            self.parse_statistics(response[0])
+        elif profile:
```

Comment:
Looks like profile can be set regardless of response. Let's revisit this if/else combination.